bundles/org.openhab.core.config.discovery.mdns/src/main/java/org/openhab/core/config/discovery/mdns/internal/MDNSDiscoveryService.java

```diff
@@ -59,7 +60,7 @@
 
     private final Set<MDNSDiscoveryParticipant> participants = new CopyOnWriteArraySet<>();
 
-    private MDNSClient mdnsClient;
+    private @NonNullByDefault({}) MDNSClient mdnsClient;
```

Comment:
Can we use constructor injection for this reference?

Reply:
I've pushed some changes that I think will work to accomplish this.